### PR TITLE
Try splitting style book into sections.

### PR DESCRIPTION
**Pull request by tellthemachines**
## What?

Split style book into sections according to the[ latest designs](https://github.com/WordPress/gutenberg/issues/53431#issuecomment-2517147624).

## Testing Instructions

1. With a block theme activated, navigate to "Styles" in the site editor. 
2. Select the style book preview by clicking the "Style book" button in the content area.
3. Navigate around global styles and see sections changing.

## Screenshots or screencast 

https://github.com/user-attachments/assets/f8195393-69a3-490d-8b85-3e73219a5286

### CODE REVIEW

**Comment by tellthemachines** [comment]
E2e failure is legit; I messed something up in the classic stylebook 😬 

fixing now

**Comment by tellthemachines** [comment]
...and just fixed another little boo-boo 😄

File: packages/edit-site/src/components/style-book/index.js [hunk 1]
```diff
@@ -405,6 +422,41 @@ export const StyleBookPreview = ( { userConfig = {}, isStatic = false } ) => {
...
+		( category ) => category.slug === previewCategory
+	);
+
+	// If there's no category definition there may be a single block.
+	const filteredExamples = categoryDefinition
+		? getExamplesByCategory( categoryDefinition, examples )
+		: {
+				examples: [
+					examples.find(
+						( example ) => example.name === previewCategory
```
Comment (ramonjd) [inline packages/edit-site/src/components/style-book/index.js:442]: `comments-pagination`, `list-item` and maybe some other blocks don't have any examples and will bork the whole page. 

`/wp-admin/site-editor.php?p=%2Fstyles&section=%2Fblocks%2Fcore%252Fcomments-pagination&preview=stylebook`

`/wp-admin/site-editor.php?p=%2Fstyles&section=%2Fblocks%2Fcore%252Flist-item&preview=stylebook`

Some other blocks are listed on https://github.com/WordPress/gutenberg/issues/64707. Also there are links to PRs where we created missing block examples.

**Review by ramonjd — COMMENTED** [review]
This is much more useful I think, thanks for getting this PR up. 

Do you think we can get rid of the scroll behavior too? Seems like we don't need it anymore, or I could be missing something.

File: packages/edit-site/src/components/style-book/index.js (hunk 1 above)
Comment (tellthemachines) [reply packages/edit-site/src/components/style-book/index.js:442]: Oh I just fixed that! Blocks without examples should now just show the all blocks page, I think that's the best we can do until we get all the examples added.

**Comment by tellthemachines** [comment]
Thanks for reviewing!

> Do you think we can get rid of the scroll behavior too? Seems like we don't need it anymore, or I could be missing something.

Do you think so? Because we still have a couple longish pages (Typography and all blocks) I thought it might be good to keep it in place for now.

Anything specific seem off about it?

**Comment by ramonjd** [comment]
> Anything specific seem off about it?

Ah, not at all. I must have been only testing with short pages. Makes sense to keep it intact for the longer pages. 🍺

**Comment by tellthemachines** [comment]
I made a couple of adjustments to the scrolling logic so that it now only scrolls long pages to the top and removed the block-specific logic as that's no longer needed with the individual block pages.

File: packages/edit-site/src/components/style-book/index.js [hunk 2]
```diff
@@ -89,35 +92,23 @@ const scrollToSection = ( anchorId, iframe ) => {
...
  * @param {string} path An internal Block Editor navigation path.
  * @return {null|{block: string}} An object containing the example to navigate to.
  */
 const getStyleBookNavigationFromPath = ( path ) => {
 	if ( path && typeof path === 'string' ) {
-		if ( path === '/' ) {
+		if (
+			path === '/' ||
+			path.startsWith( '/typography' ) ||
+			path.startsWith( '/blocks' )
```
Comment (ramonjd) [inline packages/edit-site/src/components/style-book/index.js:107]: Should `/colors` also be here? Without, there's no scroll to top

https://github.com/user-attachments/assets/80c5a27d-6c03-4765-a995-8d527a95d606

I'll tentatively add it

Comment (ramonjd) [reply packages/edit-site/src/components/style-book/index.js:107]: Done in [ad39bce](https://github.com/WordPress/gutenberg/pull/68071/commits/ad39bce976f0648e2b0fd047f2fe6b2d0c7cdec6)

**Review by ramonjd — APPROVED** [review]
This is working pretty well for me. I added scroll to top for colors as well. Seem to match the pattern. Thanks!

**Pull request merged**